python-main.js

```diff
@@ -1234,6 +1234,11 @@ function web_editor(config) {
                     }
                     return PartialFlashing.flashAsync(window.dapwrapper, image, updateProgress);
                 })
+                .catch(function(e) {
+                    alert(config.translate.alerts.error + e.message);
+                    return;
+                });
```

Comment:
I didn't notice that there is already a catch at the function return that would capture this, so we don't need to add this one:

```js

        return p.then(function() {
            $("#flashing-overlay-container").hide();

            // Send flash timing event
            var timeTaken = (new Date().getTime() - startTime);
            var details = {"flash-type": (usePartialFlashing ? "partial-flash" : "full-flash"), "event-type": "flash-time", "message": timeTaken};
            document.dispatchEvent(new CustomEvent('webusb', { detail: details }));
        })
        .catch(webusbErrorHandler)
        .finally(function() {
            // Remove event listener
            window.removeEventListener("unhandledrejection", webusbErrorHandler);
        });
```

And that one will print the error in a modal, with more instructions, which is preferable. So we should instead change the non-partial-flashing path to use that as well.

There is also the issue in that try/catch `error` variable is then called as `err`, which throws a new undefined error:

So from:
```js
try {
 var output = generateFullHex("string");
} catch(error) {
 alert(err.message);
 return;
}
```

To just:

```js
var output = generateFullHex("string");
```
Then let the promise `catch()` from the return statement deal with it.

Reply:
Cool, I think the latest commit should solve both